### PR TITLE
Add data store for user preferences

### PR DESCRIPTION
**Pull request by DouglasBruce**

### CODE REVIEW

File: app/src/main/java/com/douglasbruce/tasky/MainActivity.kt [hunk 1]
```diff
@@ -3,22 +3,52 @@ package com.douglasbruce.tasky
...
+            }
+        }
+
         splashScreen.setKeepOnScreenCondition {
-            false
+            when (uiState) {
+                Loading -> true
+                is Success -> false
+            }
         }
```
Comment (philipplackner) [inline app/src/main/java/com/douglasbruce/tasky/MainActivity.kt:52]: Couldn't you just refer to the ViewModel's `uiState` here? I don't see how the extra `uiState` initialized above and the lifecycle scope block is necessary at this point.

The `setKeepOnScreenCondition` should check the value of the state during each frame, so you shouldn't need your own variable in the activity here

File: app/src/main/java/com/douglasbruce/tasky/core/datastore/TaskyPreferencesDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package com.douglasbruce.tasky.core.datastore
```
Comment (philipplackner) [inline app/src/main/java/com/douglasbruce/tasky/core/datastore/TaskyPreferencesDataSource.kt:1]: I see, you have a package `data` and a package `datastore`. How comes, data store does not belong into `data`?

File: app/src/main/java/com/douglasbruce/tasky/MainActivity.kt (hunk 1 above)
Comment (DouglasBruce) [reply app/src/main/java/com/douglasbruce/tasky/MainActivity.kt:52]: Will simplify to refer directly to the viewModels state and remove the extra initialization and lifecycle scope

File: app/src/main/java/com/douglasbruce/tasky/core/datastore/TaskyPreferencesDataSource.kt (hunk 2 above)
Comment (DouglasBruce) [reply app/src/main/java/com/douglasbruce/tasky/core/datastore/TaskyPreferencesDataSource.kt:1]: An oversight, will move datastore into the data package

**Pull request merged**